mozilla/fx-private-relay
# MPP-3478, MPP-3455, MPP-3479: Updates for custom mask creation

This PR fixes MPP-3478, MPP-3455, and MPP-3479

# New feature description

This feature updates the custom mask creation modal for premium users by adding a tooltip for address prefix validation, an error banner to handle edge cases upon generating a mask, and a success modal panel that allows users to copy their address or enable promotions.

# Demo (if applicable)

https://github.com/mozilla/fx-private-relay/assets/59676643/7c5e7872-e58e-4a45-911f-152a7ddbb21a

# How to test
Waffle flag to enable: **custom_domain_management_redesign**

1. On a premium relay account, go to your dashboard /accounts/profile/
2. Click "Generate new mask", then your custom mask domain.
3. [Acceptance criteria] Verify the following error states upon typing your new mask prefix in the text input
* The tooltip validation checkmark and "x" symbols change according to their message.
* Edge cases: The error banner that handles edge cases shows up when you use a bad word, a prefix that starts or ends with a period/hyphen, or when you use a prefix that you already created.
4. Enter a valid prefix, and click "Generate mask"
5. [Acceptance criteria] Verify that the success modal shows.
6. [Acceptance criteria] Click on the promotions checkbox, and verify that they get updated accordingly within the UI and API request upon clicking "Done" or "Copy Mask".
7. [Acceptance criteria] When clicking "Copy Mask", verify that the copy label on the mask is shown for a few seconds.


# Checklist (Definition of Done)
